project-guidelines/quality-output/security/README.md

```diff
@@ -1,12 +1,83 @@
+
+## Client secrets
+
+- Store session IDs and auth tokens in `Secure` `HttpOnly` cookies, not `localStorage` or `sessionStorage` (also SameSite=Lax)
+- Scope cookies to a domain (and a path, if applicable)
+
+## User input
+
+- Sanitize all user input, especially the values that will be passed to `dangerouslySetInnerHTML` to prevent XSS attacks
+- Always use whitelisting over blacklisting, aka list values that are allowed instead of listing values that aren’t
+- Use CAPTCHAs on sign-in and sign-up forms to prevent password enumeration
+
+## iFrames
+
+- Don’t use iFrames unless absolutely necessary (e.g. for third-party integrations)
+- Add `sandbox` attribute to enable [extra set of restrictions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#attributes) on the iFrame content
+- Disable iFrame embedding to prevent click-jacking attacks with `X-Frame-Options` header:
+
+```bash
+X-Frame-Options: "DENY"
+```
+
+## Security headers
+
+> Use [https://securityheaders.com/](https://securityheaders.com/) to verify your security headers configuration
+
+- Use strong Content Security Policy headers ([CSP](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy)) to mitigate XSS and data injection attacks
+
+```bash
+Content-Security-Policy: default-src 'none'; script-src 'self'; img-src 'self'; style-src 'self'; connect-src 'self';
+```
+
+It’s better to start with the most restrictive settings, and start loosening them as needed (e.g. allowing unsafe-inline styles when using styled components or whitelisting more domains when using 3rd party integrations)
+
+- Disable access to browser features and APIs that aren’t used:
+
+```bash
+Feature-Policy: accelerometer 'none'; ambient-light-sensor 'none'; autoplay 'none'; camera 'none'; encrypted-media 'none'; fullscreen 'self'; geolocation 'none'; gyroscope 'none'; magnetometer 'none'; microphone 'none'; midi 'none'; payment 'none';  picture-in-picture 'none'; speaker 'none'; sync-xhr 'none'; usb 'none'; vr 'none';
+
+Permissions-Policy: accelerometer=(), ambient-light-sensor=(), autoplay=(), camera=(), encrypted-media=(), fullscreen=(), geolocation=(), gyroscope=(), magnetometer=(), microphone=(), midi=(), payment=(), picture-in-picture=(), speaker=(), sync-xhr=(), usb=(), vr=();
+```
+
+⚠️ `Feature-Policy` header is being renamed to `Permission-Policy`, so in the future only the latter will be used.
+
+- Use [HSTS](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security) header to block all attempts to downgrade HTTPS to HTTP (or WSS to WS)
+
+```bash
+Strict-Transport-Security: max-age=63072000; includeSubDomains; preload
+```
+
+- Enable [X-XSS-Protection](https://www.owasp.org/index.php/OWASP_Secure_Headers_Project#X-XSS-Protection) mode to mitigate XSS attacks
+
+```bash
+X-XSS-Protection: 1; mode=block
+```
+
+## Github
```

Comment:
I'd mention here that if you are having some Github Actions in use, it's safer to have separate "GH application" with limited scope doing those jobs. Also added bonus is that you can permit pushes to certain branches only for that account.

Example here: https://github.com/strvcom/xx-aviation-frontend/blob/develop/.github/workflows/xx-merge-reusable.yaml